graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java

```diff
@@ -725,16 +725,17 @@ public Token generateNewToken(
             @ApiParam(name = "userId", required = true) @PathParam("userId") String userId,
             @ApiParam(name = "name", value = "Descriptive name for this token (e.g. 'cronjob') ", required = true) @PathParam("name") String name,
             @ApiParam(name = "JSON Body", value = "Can optionally contain the token's TTL.", defaultValue = "{\"token_ttl\":null}") GenerateTokenTTL body) {
-        final User user = loadUserById(userId);
-        final String username = user.getName();
+        final User futureOwner = loadUserById(userId);
+        final User currentUser = getCurrentUser();
 
-        if (!isTokenCreationAllowed(user)) {
+        if (currentUser == null || !isTokenCreationAllowed(currentUser, futureOwner)) {
```

Comment:
NIT: maybe throw specific exceptions in isTokenCreationAllowed to get a more specific reason back why it failed